zoom_sesh.py

```diff
@@ -6,73 +6,97 @@
   df['year'] = [random.choice(years) for _ in range(len(df))]
 
-  df = df.iloc[:max_people]  
   person_id = list(map(str,np.arange(max_people).tolist()))
 
-  for i in person_id:
-    df[i] = np.zeros(len(df))
-    df[i+"_cnsctv"] = np.zeros(len(df))
+  #df = ZoomSesh._create_tracking_cols(df.iloc[:max_people])
+  if os.path.isdir(dir_name):
+    if overwrite:
+      shutil.rmtree(dir_name)
+      os.mkdir(dir_name)
+    else:
+      raise ValueError(f'{dir_name} already exists! Please set `overwrite` to `False`')
+  else:
+    os.mkdir(dir_name)
+
+  w = pd.ExcelWriter(f'{dir_name}/alumni.xlsx')
+  df.iloc[:max_people].to_excel(w, index=False)
+  w.save()
+  w.close()
 
-  return df.iloc[:max_people]
 
 class ZoomSesh:
   '''Object that helps organize large groups of people during a zoom call.'''
 
-  def __init__(self, filename=None, max_people=40):
-    '''Constructor
+  def __init__(self, session_directory):
+    '''Creates a ZoomSesh from student file contained in session_directory
     Args:
-      filename: location of a file containing the alumni for this session. Columns
-        TBD. If `None`, then ZoomSesh will initialize with fake data of len `max_people`.
-  
-      max_people: int specifying number of people for fake data. Does not do anything
-        unless `filename` is None.
-  
+      session_directory: folder containing important information about a session. This folder
+        *must* contain a file called 'alumni.xlsx' containing the columns 'name', 'year',
+        and 'track'.
     '''
-    if filename is not None:
-      raise NotImplementedError('This feature is not ready yet')
-    
-    else:
-      # For development, testing, debugging, etc.
-      self._alumni_history = [make_fake_data(max_people=max_people)]
+
+    if 'alumni.xlsx' not in os.listdir(session_directory):
+      raise FileNotFoundError(f"No alumni file found! Please make sure to include 'alumni.xlsx' in {session_directory}")
+      
+    alumni_file = os.path.join(session_directory, ALUMNI_FILE)
+    self._session_directory = session_directory
+    self._alumni_data = pd.read_excel(alumni_file) # DataFrame with raw data from alumni file
+
+    # Attributes for this zoom session. These are taken from the column names of the alumni file
+    # For now, these will likely just be 'name', 'year' and 'track'
+    self.attributes = [col for col in self._alumni_data.columns]
+
+    self._alumni_data = self._create_tracking_cols((self._alumni_data))
+
+    self._alumni_history = []
+    self._breakout_history = []
+
+  @staticmethod
+  def _create_tracking_cols(df):
+    df = df.copy()
+    for i in df.index:
+      df[f'{i}'] = np.zeros(len(df))
+      df[f'{i}_cnsctv'] = np.zeros(len(df))
+    return df
 
   @property
   def alumni(self):
-    '''Returns 'current' alumni matrix, which is the top matrix in the stack.'''
-    return self._alumni_history[0]
-
+    return self._alumni_data.copy()
```

Comment:
You made this change b/c the history is now being stored in the saved excel files yeah?

Reply:
Yes. I was running into issues with assigning a DataFrame. They are passed by reference and require the `.copy()`